compose_api.go

```diff
@@ -92,6 +96,45 @@ func (f StackIdentifier) String() string {
 	return string(f)
 }
 
+type ComposeStackReaders []io.Reader
+
+func (r ComposeStackReaders) applyToComposeStack(o *composeStackOptions) {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+
+	// choose directory to keep temporary files
+	// like
+	// 		/tmp/testcontainers-go/my-awesome-service-2f686f6d652f676f666f7262726f6b65
+	projectName := filepath.Base(currentDir)
+	projectHash := fmt.Sprintf("%x", fnv.New32a().Sum([]byte(currentDir)))[:32]
```

Comment:
We are using `uuid.New().String()` for new UUID. I don't mind using the project hash, but for consistency, I'd advocate for one single manner.

We could send a follow-up PR calculating the project hash everywhere in the compose code.

Reply:
Each time `uuid.New().String()` is called, a new hash will be created, new directories will be created like
/tmp/testcontainers-go/my-awesome-project-SOME_NEW_HASH_1/
/tmp/testcontainers-go/my-awesome-project-SOME_NEW_HASH_2/
/tmp/testcontainers-go/my-awesome-project-SOME_NEW_HASH_3/
So we will litter the machine with directories. This reason why I did not use uuid lib.
I think better use hash func that always returns same hash for project name/path.

Reply:
Maybe each temporary docker-compose-???.yml file should contains hash also. For situation when developer runs couple of functional tests in parallel.

Reply:
Indeed. Let's keep it deterministic then. Thanks for the clarification